src/bde_copy_utils.cpp

```diff
@@ -990,7 +990,15 @@ int geometry_field::write_offset_geom( char *sp, data_writer *out )
 
 data_writer *file_data_writer::open( char *fname, bool append )
 {
-    FILE *f = fopen(fname,append ? "ab" : "wb" );
+    FILE *f;
+    if( strcmp(fname, "-") == 0)
```

Comment:
I would rather handle this outside of the function. If you know it's stdout don't call file_data_writer::open at all ?

Reply:
I have removed this section from the file_data_writer::open function and instead have created a separate function to call if the output is going to stdout. Is this what you were meaning?